### PR TITLE
compiling.rst: add explanation for CLD2 dependency

### PR DESCRIPTION
**Pull request by anatoli26**
Closes 3102 (1 pending)

### CODE REVIEW

File: docsrc/imap/developer/compiling.rst [hunk 1]
```diff
@@ -178,6 +178,10 @@ Other
     installing SSL certificates."
     `ClamAV`_, libclamav-dev, clamav-devel, "no", "It is used by
     **cyr_virusscan**, otherwise not needed."
+    `CLD2`_, libcld2-dev, cld2-devel, "yes/no", "Compact Language Detector 2
+    (probabilistically detects over 80 languages in Unicode UTF-8 text, either
+    plain text or HTML/XML). Required for **Xapian** (``--enable-xapian``),
```
Comment (rsto) [inline docsrc/imap/developer/compiling.rst:183]: It is required for Xapian if the search_index_language option in imapd.conf is set. Thanks!

Comment (anatoli26) [reply docsrc/imap/developer/compiling.rst]: I just added the clarification about `search_index_language`.

Comment (elliefm) [reply docsrc/imap/developer/compiling.rst]: @rsto What happens if `search_index_language` is enabled on an installation that was compiled without CLD2?

Comment (anatoli26) [reply docsrc/imap/developer/compiling.rst]: I guess we get the error @brong mentioned in the related issue: "IOERROR: Xapian: language indexing requires CLD2 library"

**Comment by anatoli26** [comment]
@elliefm, any feedback on this PR?

**Comment by elliefm** [comment]
I think I'd prefer to leave the detail about `search_index_language` out, and leave it so that we say it's needed for any Xapian (regardless of that setting).

My general preference is that changing a setting in imapd.conf shouldn't suddenly require a recompile with new dependencies -- you either have the feature built in and can enable/disable/customise it from imapd.conf, or you don't have the feature built in, and any settings for it are ignored.

I think you could probably reset your branch to just the first commit (discarding the ", p2" one), force push, and we'd be in a good place.

**Comment by anatoli26** [comment]
@elliefm, I just force-pushed the undo of the last commit as you suggested.

Now, what do you think about adding a check for cld2 as a required dependency in configure when Xapian is enabled?

**Comment by elliefm** [comment]
I think I previously didn't want to do that, but now that I understand that this dependency's presence determines whether an imapd.conf setting works or not, I've changed my mind and I think that's the right way to go.  I'll add a similar note to the main issue.

**Pull request merged**